cardea/problem_definition/readmission.py

```diff
@@ -0,0 +1,144 @@
+
+            Args:
+            es: fhir entityset.
+            target_label: The target label of the prediction problem.
+            target_entity: The entity name in which it contains the target label.
+
+            Returns:
+            Updated entityset with the generated label.
+
+            Raises:
+            ValueError: An error occurs if the target label cannot be generated.
+            """
+        generate_from = 'Period'
+        end = 'end'
+
+        try:
+            if self.check_target_label(
+                es,
+                generate_from,
+                end) and self.check_target_label(es,
+                                                 self.target_entity,
+                                                 'period'):
+
+                try:
+                    self.check_target_label_values(
+                        es,
+                        generate_from,
+                        end)
+
+                    entity_set_df = es[self.target_entity].df
+                    generate_df = es[generate_from].df
+                    merged_df = pd.merge(entity_set_df, generate_df,
+                                         left_on='period', right_on='object_id')
+
+                    generated_target_label = []
+                    encounter_identifier = []
+
+                    for patient in set(merged_df['subject']):
+                        patient_visits = merged_df[merged_df['subject'] == patient]
+                        inital_date = patient_visits[end].iloc[0]
+
+                        encounter_identifier.append(patient_visits['identifier'].iloc[0])
+                        generated_target_label.append(False)  # first visit
+
+                        if len(patient_visits) != 1:
+                            for visit_date, encounter_id in zip(patient_visits[end][1:],
+                                                                patient_visits['identifier'][1:]):
+
+                                visit_range = visit_date - inital_date
+                                inital_date = visit_date
+
+                                if visit_range.days <= 30:
+                                    generated_target_label.append(True)
+                                    encounter_identifier.append(encounter_id)
+
+                                else:
+                                    generated_target_label.append(False)
+                                    encounter_identifier.append(encounter_id)
+
+                    generated_labels = pd.DataFrame(
+                        {self.target_label: generated_target_label,
+                         'identifier': encounter_identifier})
+                    updated_target_entity = pd.merge(entity_set_df,
+                                                     generated_labels,
+                                                     on='identifier')
+
+                    es = es.entity_from_dataframe(entity_id='Encounter',
```

Comment:
@ShahadThobaiti Any reason we don't change this to `self.target_entity`
```suggestion
                    es = es.entity_from_dataframe(entity_id=self.target_entity,
```